kernel/threads.c

```diff
@@ -51,6 +51,7 @@ static tcb *create_thread (void *text, size_t length, vmcb *vm_space) {
       tcbs[i].state = TS_EXIST;
       tcbs[i].stack_top = &kernel_stack[0];
       tcbs[i].fpu_state = THREAD_FPU_STATE_INACTIVE;
+      tcbs[i].ipc_state = IPC_NOT_RECEIVING;
       total_threads++;
```

Comment:
Recalculating the number of tasks each time in 'choose_task' is fine, but can you delete 'total_threads' then?

(Alternatively, you could have the 'send to the kernel' case also reach in and decrement total_threads... but that might get hairy once you can also kill daemons.  So statelessly checking every time is pretty reasonable.)

Reply:
I didn't realize that was unused now - gladly deleted :)